update-common/src/artifacts/update_plan.rs

```diff
@@ -317,6 +320,44 @@ impl<'a> UpdatePlanBuilder<'a> {
             },
         )?;
 
+        // We need to get all the signing information now to properly check
+        // version at builder time (builder time is not async)
+        let image_a_stream = rot_a_data
+            .reader_stream()
+            .await
+            .map_err(RepositoryError::CreateReaderStream)?;
+        let mut image_a = Vec::with_capacity(rot_a_data.file_size());
+        tokio_util::io::StreamReader::new(image_a_stream)
+            .read_to_end(&mut image_a)
+            .await
+            .expect("failed to read RoT image a");
```

Comment:
I don't think we can `.expect()` these - `reader_stream()` returns streams that are backed by `tokio::fs::File`s, which can fail at runtime. Probably need something like a 

```rust
RepositoryError::ReadExtractedArchive {
    artifact: ArtifactHashId,
    #[source]
    error: std::io::Error,
}
```

variant to use here?